### PR TITLE
fix `Wrong handler is called for unautorized cross-domain request` (close #1263)

### LINKED ISSUE #1263: Wrong handler is called for unautorized cross-domain request
For unauthorized cross-domain request (don't pass CORS rules) we call load handler instead of an error handler.

Script for reproduce:
With proxing, a browser console will be contain 'loadHandler is called' message.
Without - 'errorHandler is called';
```js
    var xhr = new XMLHttpRequest();

   // http://cross-domain.com/ should not respond CORS headers.
    xhr.open('GET', 'http://cross-domain.com/', true);
   
    
    function errorHandler () {
        console.log('errorHandler is called');
    }

    function loadHandler () {
       console.log('loadHandler is called');
    }

    xhr.addEventListener('load', loadHandler);
    xhr.addEventListener('abort', errorHandler);
    xhr.addEventListener('error', errorHandler);
    xhr.addEventListener('timeout', errorHandler);

    xhr.send();

```

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

**Review by miherlosev — APPROVED** [review]

**Review by Farfurix — APPROVED** [review]

File: src/session/events/info.ts [hunk 1]
```diff
@@ -36,7 +35,7 @@ export class ResponseInfo {
         this.requestId  = ctx.requestId;
         this.headers    = ctx.destRes.headers;
         this.body       = ctx.nonProcessedDestResBody;
-        this.statusCode = ctx.isSameOriginPolicyFailed ? SAME_ORIGIN_CHECK_FAILED_STATUS_CODE : ctx.destRes.statusCode;
+        this.statusCode = ctx.isSameOriginPolicyFailed ? 0 : ctx.destRes.statusCode;
```
Comment (LavrovArtem) [inline src/session/events/info.ts:38]: !!! Maybe, we always need to return `ctx.destRes.statusCode`.

File: src/typings/client.d.ts [hunk 2]
```diff
@@ -19,6 +19,10 @@ export interface HammerheadInitSettings {
     windowId: string;
 }
 
+declare const TextRange: {
```
Comment (LavrovArtem) [inline src/typings/client.d.ts:22]: delete it

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Comment by LavrovArtem** [comment]
[testcafe-hammerhead-17.2.0.zip](https://github.com/DevExpress/testcafe-hammerhead/files/5709867/testcafe-hammerhead-17.2.0.zip)

**Review by miherlosev — APPROVED** [review]

**Review by Farfurix — APPROVED** [review]

**Comment by LavrovArtem** [comment]
don't merge it

File: src/utils/headers.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+// -------------------------------------------------------------
+// WARNING: this file is used by both the client and the server.
+// Do not use any browser or node-specific API!
+// -------------------------------------------------------------
+
+import BUILTIN_HEADERS from '../request-pipeline/builtin-header-names';
+
+const AUTHENTICATE_PREFIX  = '~~~this prefix added by testcafe for the native dialog preventing~~~ ';
+const AUTHORIZATION_PREFIX = '~~~this prefix added by testcafe for the authorization controlling~~~ ';
```
Comment (LavrovArtem) [inline src/utils/headers.ts:9]: check it please

Comment (titerman) [inline src/utils/headers.ts:9]:
```suggestion
const AUTHENTICATE_PREFIX  = '~~~TestCafe added this prefix to hide the authentication dialog box~~~ ';
const AUTHORIZATION_PREFIX = '~~~TestCafe added this prefix to control the authorization flow~~~ ';
```

**Review by titerman — CHANGES_REQUESTED** [review]

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Pull request merged**